File: examples/python/src/exampleVaem.py
from time import sleep
import logging
import asyncio

from driver.VaemDriver import vaemDriver
from driver.dataTypes import VaemConfig


if __name__ == "__main__":
    vaemConfig = VaemConfig('192.168.8.118', 502, 0)

    try:
        vaem = vaemDriver(vaemConfig, logger=logging)
    except Exception as e:
        print(e)
    async def func():
        vaem.init()
        print(vaem.read_status())
        await vaem.select_valve(3)
        print(vaem.read_status())
        await vaem.deselect_valve(3)
        print(vaem.read_status())
        await vaem.select_valve(7)
        print(vaem.read_status())
        await vaem.deselect_valve(7)
        print(vaem.read_status())

    loop = asyncio.get_event_loop()
    loop.run_until_complete(func())

File: examples/java/src/exampleVaem.java
import driver.VaemDriver;

import static java.lang.Thread.sleep;

public class exampleVaem {
    public static void main(String[] args) throws InterruptedException {
        int[] valveData = {500, 0, 0, 0, 0, 0, 0, 0};
        VaemDriver vm = new VaemDriver("192.168.0.214", 502);
        vm.init();
        vm.configureValves(valveData);

        while (true) {
            sleep(1000);
            vm.openValve();
            sleep(1000);
            vm.readStatus();
            vm.closeValve();
        }



    }

}
